build_tools/kokoro/gcp_ubuntu/cmake/linux/riscv64/test.sh

```diff
@@ -18,41 +18,64 @@ export PS4='[$(date -u "+%T %Z")] '
 # dylib-llvm-aot bytecode codegen.
 export RISCV_TOOLCHAIN_ROOT=${RISCV_RV64_LINUX_TOOLCHAIN_ROOT?}
 
-# Run the binaries under QEMU.
-echo "Test simple_embedding binaries"
-pushd "${BUILD_RISCV_DIR?}/iree/samples/simple_embedding" > /dev/null
 
-"${QEMU_RV64_BIN?}" -cpu rv64,x-v=true,x-k=true,vlen=256,elen=64,vext_spec=v1.0 \
--L "${RISCV_TOOLCHAIN_ROOT?}/sysroot" simple_embedding_dylib
+function generate_dylib_vmfb {
+  local target="${1}"; shift
+  local translate_arg=(
+    -iree-mlir-to-vm-bytecode-module -iree-hal-target-backends=dylib-llvm-aot
+    -iree-llvm-embedded-linker-path=${BUILD_HOST_DIR?}/install/bin/lld
+    -iree-llvm-target-triple=riscv64
+    -iree-llvm-target-cpu=generic-rv64
+    -iree-llvm-target-abi=lp64d
+  )
+  if [[ "${target}" == "tosa" ]]; then
+    local input_file="${1}"; shift
+    iree-import-tflite -o "${BUILD_RISCV_DIR?}/tosa.mlir" "${input_file}"
+    translate_arg+=(
+      -iree-input-type=tosa
+      -iree-llvm-target-cpu-features="+m,+a,+f,+d,+c"
+      "${BUILD_RISCV_DIR?}/tosa.mlir"
+    )
+  elif [[ "${target}" == "tosa-rvv" ]]; then
+    local input_file="${1}"; shift
+    iree-import-tflite -o "${BUILD_RISCV_DIR?}/tosa.mlir" "${input_file}"
+    translate_arg+=(
+      -iree-input-type=tosa
+      -iree-llvm-target-cpu-features="+m,+a,+f,+d,+c,+v"
+      -riscv-v-fixed-length-vector-lmul-max=8
+      -riscv-v-vector-bits-min=256
+      -riscv-v-fixed-length-vector-elen-max=64
+      "${BUILD_RISCV_DIR?}/tosa.mlir"
+    )
+  elif [[ "${target}" == "mhlo" ]]; then
+    translate_arg+=(
+      -iree-input-type=mhlo
+      -iree-llvm-target-cpu-features="+m,+a,+f,+d,+c"
+    )
+  fi
+  "${BUILD_HOST_DIR?}/install/bin/iree-translate" "${translate_arg[@]}" "$@"
+}
 
-"${QEMU_RV64_BIN?}" -cpu rv64,x-v=true,x-k=true,vlen=256,elen=64,vext_spec=v1.0 \
--L "${RISCV_TOOLCHAIN_ROOT?}/sysroot" simple_embedding_embedded_sync
+PIP_INSTALL_DIR="${BUILD_RISCV_DIR?}/pip/"
 
-"${QEMU_RV64_BIN?}" -cpu rv64,x-v=true,x-k=true,vlen=256,elen=64,vext_spec=v1.0 \
--L "${RISCV_TOOLCHAIN_ROOT?}/sysroot" simple_embedding_vmvx_sync
+mkdir -p "${PIP_INSTALL_DIR}"
+pip install --prefix=${PIP_INSTALL_DIR} lit iree-tools-tflite
```

Comment:
This moves from testing IREE at the current commit to using artifacts from a previous release. In general, we make no guarantees of forward/backward compatibility of the interchange formats here (eventually we will, but not right now). The only build we have doing that sort of thing right now is the samples build, which only runs postsubmit and is not generally supported (basically, Scott just monitors it). In contrast, this build runs on presubmit and sends notification emails to our build cop rotation when it fails. I don't want to turn that into something that isn't based on the current commit. It also adds multiple network access requirements. This already got added to the tflite build, but it still makes me very sad. There's a place for that sort of test, but I don't think it's here.

I'm looking to revamp all our CI stuff, which would allow something like: one job fetches files with caching and retry strategies, one job builds integrations bazel binaries, and then the downstream things like this can make use of those while itself staying hermetic.

Reply:
The trickiest part is `iree-tools-tflite` is not built by default in host, and it is required by the `.tflite` input. If there is a way of converting `.tflite` to `.mlir` using the artifact built by the host, I am happy to switch to it.